### PR TITLE
Configure jest, RTL & add breakdown card unit test

### PR DESCRIPTION
**Pull request by alfonsomthd**
- Added jest configuration + React Testing Library (RTL) to run unit tests.
- Updated GitHub action to include `yarn test-coverage`.
- Added 1st unit test: capacity-breakdown-card.
- NOTE: jest config is kept as a JavaScript file because using a TypeScript file requires upgrading `ts-node` that in turn
  makes `yarn build` too slow (>200 s).

Signed-off-by: Alfonso Martínez <almartin@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,6 +148,29 @@
...
+    "transformIgnorePatterns": [
+      "<rootDir>/node_modules/(?!(@openshift-console)/)"
+    ],
+    "testPathIgnorePatterns": [
+      "<rootDir>/cypress/",
+      "<rootDir>/node_modules/",
+      "<rootDir>/plugins/"
+    ],
+    "testRegex": ".*\\.spec\\.[t]sx?$"
+  },
```
Comment (SanjalKatiyar) [inline package.json:173]: for later, when we mark this PR up for review...
let's have separate config file for jest as well...

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  testEnvironment: 'jsdom',
+  transform: {
+    '\\.[jt]sx?$': '@swc/jest',
+  },
+  transformIgnorePatterns: ['<rootDir>/node_modules/(?!(@openshift-console)/)'],
+  testPathIgnorePatterns: [
+    '<rootDir>/cypress/',
+    '<rootDir>/node_modules/',
+    '<rootDir>/plugins/',
+  ],
```
Comment (SanjalKatiyar) [inline jest.config.js:20]: Is it required even if we have added `roots: ['<rootDir>/packages/'],` ??

File: package.json [hunk 3]
```diff
@@ -15,6 +15,9 @@
     "cypress-merge": "mochawesome-merge ./gui-test-screenshots/cypress_report*.json > ./gui-test-screenshots/cypress.json",
     "cypress-generate": "marge -o ./gui-test-screenshots/ -f cypress-report -t 'OpenShift Console Cypress Test Results' -p 'OpenShift Cypress Test Results' --showPassed false --assetsDir ./gui-test-screenshots/cypress/assets ./gui-test-screenshots/cypress.json",
     "cypress-postreport": "yarn cypress-merge && yarn cypress-generate",
+    "test": "jest",
+    "test-silent": "JEST_SILENT_REPORTER_DOTS=true jest --reporters jest-silent-reporter",
```
Comment (SanjalKatiyar) [inline package.json:19]: where are these 2 commands are being used ??

File: packages/shared/table/table.tsx [hunk 4]
```diff
@@ -80,7 +80,7 @@ const Table: React.FC<TableProps> = React.memo(
     return (
       <TableComposable aria-label={ariaLabel} translate={null}>
         <Thead translate={null}>
-          <Tr translate={null}>{...headerColumns}</Tr>
+          <Tr translate={null}>{headerColumns}</Tr>
```
Comment (SanjalKatiyar) [inline packages/shared/table/table.tsx:83]: I wonder why this component was working correctly so far (maybe typescript is doing some magic) !! It's a good catch.

File: tsconfig.json [hunk 5]
```diff
@@ -12,5 +12,6 @@
     "sourceMap": true,
     "noEmitOnError": false
   },
+  "exclude": ["**/*.spec.ts", "**/*.spec.tsx"],
```
Comment (SanjalKatiyar) [inline tsconfig.json:15]: If test files are of type `.ts` `.tsx` then why are we excluding them from ts config file ?? Maybe I am missing something ??

File: packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+jest.mock('@openshift-console/dynamic-plugin-sdk/lib/utils/flags', () => ({
+  useFlag: () => true,
+}));
+
+jest.mock(
+  '@odf/shared/hooks/custom-prometheus-poll/custom-prometheus-poll-hook',
+  () => ({
+    useCustomPrometheusPoll: () => [true, false, false],
+  })
+);
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx:24]: can we add it to setup file as well ??

File: packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+jest.mock('@openshift-console/dynamic-plugin-sdk/lib/extensions', () => {
+  const originalModule = jest.requireActual<
+    typeof import('@openshift-console/dynamic-plugin-sdk/lib/extensions')
+  >('@openshift-console/dynamic-plugin-sdk/lib/extensions');
+  return {
+    ...originalModule,
+    useK8sWatchResource: () => [true, false, false],
+  };
+});
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx:13]: same this as well to setup file ??
these utilities will be shared by multiple test files...

File: jest.config.js (hunk 2 above)
Comment (alfonsomthd) [reply jest.config.js:20]: I think we can drop this as long we keep `roots` (or on reverse). For simplicity, we can keep `roots`.

File: package.json (hunk 3 above)
Comment (alfonsomthd) [reply package.json:19]: These scripts are to be run on CLI  for running the unit tests. After a conversation with @bipuladh , we decided for now to not include the tests in pre-commit, so they are run in CI job as `yarn test-coverage`.
Any developer can run `yarn test` to check its code locally and `yarn test-silent` is meant to run your tests without your terminal being cluttered by the output (at Ceph Dashboard we have lots of unit tests and this is very convenient!).

File: tsconfig.json (hunk 5 above)
Comment (alfonsomthd) [reply tsconfig.json:15]: We're excluding them from the build (you can check which files are to be included in the build by running `yarn tsc --showConfig` and checking "files").
Test files are to be used only by Jest (or any test library).

File: packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx (hunk 6 above)
Comment (alfonsomthd) [reply packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx:24]: The thing is for this unit test it's fine to return `[true, false, false]` but I'm not sure if for other tests that payload is OK or even if we need that mock for them in the first place...
I prefer to not promote a mock to the setup file until really needed (or there is a general consensus to do it).
@bipuladh Thoughts?

File: packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx (hunk 7 above)
Comment (alfonsomthd) [reply packages/ocs/dashboards/object-service/capacity-breakdown/capacity-breakdown-card.spec.tsx:13]: Same as above.

File: package.json (hunk 3 above)
Comment (SanjalKatiyar) [reply package.json:19]: ack !!

File: tsconfig.json (hunk 5 above)
Comment (SanjalKatiyar) [reply tsconfig.json:15]: make sense... thanks !!

File: package.json (hunk 3 above)
Comment (SanjalKatiyar) [reply package.json:19]: created this: https://github.com/red-hat-storage/odf-console/issues/546 for later.

**Pull request merged**